### PR TITLE
k8s-infra: Add azure credential plugin

### PR DESCRIPTION
**Pull request by ameukam**
Add https://github.com/Azure/kubelogin so we can use it for authn to AKS clusters from Prow Control plane.

### CODE REVIEW

**Comment by ameukam** [comment]
cc @BenTheElder @upodroid @dims 

/hold

File: images/git-custom-k8s-auth/Dockerfile [hunk 1]
```diff
@@ -12,26 +12,32 @@
...
 
+RUN apt update && apt install -y unzip && curl -fsSLO https://github.com/Azure/kubelogin/releases/download/v${AZURE_KUBELOGIN_VERSION}/kubelogin-linux-amd64.zip \
+ && unzip -j kubelogin-linux-amd64.zip -d / \
+ && rm kubelogin-linux-amd64.zip
 
 # Update the date below (check https://hub.docker.com/_/alpine/tags) to get the
 # alpine edge base image, which has a faster release cadence than alpine:latest
 # (which tracks alpine stable), to pick up security patches more quickly.
-FROM alpine:20231219
+FROM alpine:20240606
```
Comment (BenTheElder) [inline images/git-custom-k8s-auth/Dockerfile:32]: TODO: switch this, cc @dims 

I would recommend something like how we did distroless-iptables

Comment (ameukam) [reply images/git-custom-k8s-auth/Dockerfile:32]: what's the issue with alpine ?

Comment (dims) [reply images/git-custom-k8s-auth/Dockerfile:32]: @ameukam https://github.com/kubernetes/enhancements/blob/master/keps/sig-release/1729-rebase-images-to-distroless/README.md#type-3-alpine-based-images

Comment (dims) [reply images/git-custom-k8s-auth/Dockerfile:32]: let's add a TODO and/or an issue to switch this off of alpine for later

Comment (BenTheElder) [reply images/git-custom-k8s-auth/Dockerfile:32]: (also we have DNS resolver flakes when using a less mature libc)

Comment (ameukam) [reply images/git-custom-k8s-auth/Dockerfile:33]: TODO added

**Review by BenTheElder — COMMENTED** [review]
/lgtm
/approve

**Comment by BenTheElder** [comment]
I think we also have to migrate this build to k8s-infra? lots of fun 🙃

**Pull request merged**